pvlib/pvsystem.py

```diff
@@ -2037,6 +2051,108 @@ def snlinverter(v_dc, p_dc, inverter):
+        A scalar or pandas series of DC powers, in watts, which are provided
+        as input to the inverter. If Vdc and Pdc are vectors, they must be
+        of the same size. p_dc must be >= 0. (W)
+
+    inverter : dict-like
+        A dict-like object defining the inverter to be used, giving the
+        inverter performance parameters according to the model
+        developed by Anton Driesse [1].
+        A set of inverter performance parameters may be loaded from the
+        supplied data table using retrievesam.
+        See Notes for required keys.
+
+    Returns
+    -------
+    ac_power : numeric
+        A numpy array or pandas series of modeled AC power output given the
+        input DC voltage, v_dc, and input DC power, p_dc. When ac_power would
+        be greater than pac_max, it is set to p_max to represent inverter
+        "clipping". When ac_power would be less than -p_nt (energy consumed
+        rather  than produced) then ac_power is set to -p_nt to represent
+        nightly power losses. ac_power is not adjusted for maximum power point
+        tracking (MPPT) voltage windows or maximum current limits of the
+        inverter.
+
+    Notes
+    -----
+
+    Required inverter keys are:
+
+    ======   ============================================================
+    Column   Description
+    ======   ============================================================
+    p_nom    The nominal power value used to normalize all power values,
+             typically the DC power needed to produce maximum AC power output,
+             (W).
+
+    v_nom    The nominal DC voltage value used to normalize DC voltage values,
+             typically the level at which the highest efficiency is achieved,
+             (V).
+
+    pac_max  The maximum AC output power value, used to clip the output if
+             needed, (W).
+
+    ce_list  This is a list of 9 coefficients that capture the influence
+             of input voltage and power on inverter losses, and thereby
+             efficiency.
+
+    p_nt     ac-power consumed by inverter at night (night tare) to maintain
+             circuitry required to sense PV array voltage, (W).
+    ======   ============================================================
+
+    References
+    ----------
+    [1] Beyond the Curves: Modeling the Electrical Efficiency
+        of Photovoltaic Inverters, PVSC 2008, Anton Driesse et. al.
+
+    See also
+    --------
+    sapm
+    singlediode
+    '''
+
+    p_nom = inverter['Pnom']
+    v_nom = inverter['Vnom']
+    pac_max = inverter['Pacmax']
+    p_nt = inverter['Pnt']
+    ce_list = inverter['ADRCoefficients']
+
+    pdc = p_dc/p_nom
+    vdc = v_dc/v_nom
+    poly = np.array([pdc**0, pdc, pdc**2, vdc-1, pdc*(vdc-1),
+                     pdc**2*(vdc-1), 1/vdc-1, pdc*(1./vdc-1),
+                     pdc**2*(1./vdc-1)])
+
+    p_loss = np.dot(np.array(ce_list), poly)
+    ac_power = p_nom * (pdc-p_loss)
+
+    p_nt = -1*np.absolute(p_nt)
```

Comment:
That was just background info.  The absolute() shouldn't be needed, but makes it more robust.